src/main/scala/org/broadinstitute/hail/keytable/KeyTable.scala

```diff
@@ -420,4 +420,39 @@ case class KeyTable(rdd: RDD[(Annotation, Annotation)], keySignature: TStruct, v
 
     KeyTable(newRDD, keySignature, valueSignature)
   }
+
+  def explode(columnNames: Array[String]): KeyTable = {
```

Comment:
I think I would have written explode to take a single column, and then have the version that takes multiple columns to call the single-column version in a loop.  You deal with the looping in a number of places below and you can simplify the code overall by pulling the out to a single loop outside.